azure-client-runtime/src/main/java/com/microsoft/azure/v2/PollStrategy.java

```diff
@@ -0,0 +1,77 @@
+/**
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License. See License.txt in the project root for
+ * license information.
+ */
+
+package com.microsoft.azure.v2;
+
+import com.microsoft.rest.v2.http.HttpRequest;
+import com.microsoft.rest.v2.http.HttpResponse;
+import rx.Single;
+
+import java.io.IOException;
+
+/**
+ * An abstract class for the different strategies that an OperationStatus can use when checking the
+ * status of a long running operation.
+ */
+abstract class PollStrategy {
+    private long delayInMilliseconds;
+
+    /**
+     * Get the number of milliseconds to delay before sending the next poll request.
+     * @return The number of milliseconds to delay.
+     */
+    final long delayInMilliseconds() {
+        return delayInMilliseconds;
+    }
+
+    /**
+     * Set the delay in milliseconds to 0.
+     */
+    final void clearDelayInMilliseconds() {
+        this.delayInMilliseconds = 0;
+    }
+
+    /**
+     * Update the delay in milliseconds from the provided HTTP poll response.
+     * @param httpPollResponse The HTTP poll response to update the delay in milliseconds from.
+     */
+    final void updateDelayInMillisecondsFrom(HttpResponse httpPollResponse) {
+        final String retryAfterSecondsString = httpPollResponse.headerValue("Retry-After");
+        if (retryAfterSecondsString != null && !retryAfterSecondsString.isEmpty()) {
+            try {
+                final long responseDelayInMilliseconds = Long.valueOf(retryAfterSecondsString) * 1000;
+                delayInMilliseconds = Math.max(responseDelayInMilliseconds, AzureProxy.defaultDelayInMilliseconds());
```

Comment:
why we take max out of these two? Should we use `AzureProxy.defaultDelayInMilliseconds()` as delay value only if "Retry-After" header not present in the response?

Reply:
I was thinking of the scenario where a user sets the AzureProxy.defaultDelayInMilliseconds to 30 seconds, but the Retry-After header says 3 seconds. In that case, the user is in no hurry to poll again, so we can wait until the user specified 30 seconds. However, if the user specifies 30 seconds but the Retry-After header says 60 seconds, then we should respect Azure's wishes and not poll for another 60 seconds. Does this makes sense? Or should we always respect Azure's wishes?

Reply:
ah.. got it now, didn't know that user can configure `defaultDelayInMilliseconds`. Now the logic make sense. If user don't set it then library uses a default value - right?

Reply:
Correct.